Regal "panics" when comments are added to the starting brace and first field of an object

Hello,

I noticed this issue while working with the OPA Visual Studio Code Extension: https://openpolicyagent.slack.com/archives/CBR63TK2A/p1743546027512569

Regal terminates with a `panic: overwriting non-nil beforeEnd` message when comments are added to the starting brace and first field of an object in an object assignment, leading to a stack trace that eventually leads to the termination of the OPA extension with a `Connection to server got closed. Server will not be restarted.` message. If only one or the other comment is present, it does not terminate. This only appears to occur when both are present.

```
package policies.panicexample

create_token :=
        { # comment
            "header": "header" # comment
        }
```

Here is a full stack trace; the error also occurs using the `regal lint` command on the offending file.

```
panic: overwriting non-nil beforeEnd

goroutine 16 [running]:
github.com/open-policy-agent/opa/v1/format.(*writer).beforeLineEnd(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:1525
github.com/open-policy-agent/opa/v1/format.(*writer).insertComments(0x14001a27a70, {0x14000b941e0, 0x1, 0x18?}, 0x14001a139e0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:652 +0x26c
github.com/open-policy-agent/opa/v1/format.(*writer).writeTermParens(0x14001a27a70, 0x0, 0x14001205680, {0x14000b941e0?, 0x1?, 0x14001205860?})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:838 +0x44
github.com/open-policy-agent/opa/v1/format.(*writer).writeTerm(0x0?, 0x14000f419b0?, {0x14000b941e0?, 0x10463b8f4?, 0x105ddaba0?})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:834 +0x34
github.com/open-policy-agent/opa/v1/format.(*writer).writeObject.(*writer).objectWriter.func2({0x105e0ea60?, 0x14000f419b0?}, {0x14000b941e0, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:1216 +0x168
github.com/open-policy-agent/opa/v1/format.(*writer).writeIterableLine(0x14001a27a70, {0x14000f419d0, 0x1, 0x14001a13b00?}, {0x14000b941e0?, 0x14001a13da0?, 0x14001b46fc8?}, 0x14001b47078)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:1201 +0x11c
github.com/open-policy-agent/opa/v1/format.(*writer).writeIterable(0x14001a27a70, {0x14000f419c0?, 0x14001b47068?, 0x104634634?}, 0x0?, 0x14001a13da0, {0x14000b941e0, 0x1, 0x1}, 0x14001b47078)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:1177 +0x170
github.com/open-policy-agent/opa/v1/format.(*writer).writeObject(0x14001a27a70, {0x106011fa0, 0x1400115bf00}, 0x14001a13b00, {0x14000b941e0, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:1006 +0x148
github.com/open-policy-agent/opa/v1/format.(*writer).writeTermParens(0x14001a27a70, 0x0, 0x14001205668, {0x14000f41980?, 0x14000f41980?, 0x14001205800?})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:847 +0x560
github.com/open-policy-agent/opa/v1/format.(*writer).writeTerm(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:834
github.com/open-policy-agent/opa/v1/format.(*writer).writeHead(0x14001a27a70, 0x14000531b80, 0x0, 0x1, {0x14000f41980, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:640 +0x5d8
github.com/open-policy-agent/opa/v1/format.(*writer).writeRule(0x14001a27a70, 0x14001a13c80, 0x2?, {0x14000f41980, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:460 +0x190
github.com/open-policy-agent/opa/v1/format.(*writer).writeRules(0x14001a27a70, {0x14000b941c0, 0x1, 0x0?}, {0x14000f41970?, 0x2?, 0x14001b47488?})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:435 +0x70
github.com/open-policy-agent/opa/v1/format.(*writer).writeModule(0x14001a27a70, 0x14001a27950?)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:386 +0x1f0
github.com/open-policy-agent/opa/v1/format.AstWithOpts({0x105f59860?, 0x14001a278c0?}, {0x1?, 0x0?, 0x14001b47a40?, 0x0?})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:263 +0x79c
github.com/open-policy-agent/opa/v1/format.SourceWithOpts({0x0, 0x0}, {0x1400180dd00?, 0x14001a27f80?, 0x10?}, {0x10?, 0x14001b47a58?, 0x14001b47a40?, 0x68?})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/format/format.go:94 +0x1b4
github.com/styrainc/regal/pkg/builtins.RegalIsFormatted({{0x106001040, 0x140016f8190}, {0x106007650, 0x140003a2620}, {0x105ff6ba0, 0x1400009c100}, 0x14000c6fed8, {0x105ff9f20, 0x14000b0d95c}, 0x0, ...}, ...)
        /home/runner/work/regal/regal/pkg/builtins/builtins.go:153 +0x124
github.com/styrainc/regal/pkg/builtins.init.Function2.func3.1()
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/rego/rego.go:759 +0x6c
github.com/open-policy-agent/opa/v1/rego.memoize(_, {{0x106001040, 0x140016f8190}, {0x106007650, 0x140003a2620}, {0x105ff6ba0, 0x1400009c100}, 0x14000c6fed8, {0x105ff9f20, 0x14000b0d95c}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/rego/rego.go:814 +0x100
github.com/styrainc/regal/pkg/builtins.init.Function2.func3({{0x106001040, 0x140016f8190}, {0x106007650, 0x140003a2620}, {0x105ff6ba0, 0x1400009c100}, 0x14000c6fed8, {0x105ff9f20, 0x14000b0d95c}, 0x0, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/rego/rego.go:759 +0xa0
github.com/open-policy-agent/opa/v1/topdown.(*evalBuiltin).eval(0x14001a477c0, 0x14001205278)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1949 +0x400
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalCall(0x140002c2e00, {0x140006a9080, 0x3, 0x3}, 0x14001205278)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:961 +0x83c
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalStep(0x140002c2e00, 0x14000f416e0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:502 +0x2f4
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalExpr(0x140002c2e00, 0x140018da300)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:407 +0x21c
github.com/open-policy-agent/opa/v1/topdown.(*eval).eval(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:356
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalNot(0x140002c3800, 0x14000f416d0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:569 +0x318
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalStep(0x140008fddc3?, 0x10511f24b?)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:416 +0x6f8
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalExpr(0x140002c3800, 0x14001a27440)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:407 +0x21c
github.com/open-policy-agent/opa/v1/topdown.(*eval).next(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:208
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalExpr.func2(0x140002c3800)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:408 +0x30
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalStep.func5()
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:499 +0x28
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyValues(0x140002c3800, 0x14000377d58, 0x14001379bc0, 0x140018da2d0, 0x140018da2d0, 0x14001205260)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1157 +0x494
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunify(0x140002c3800, 0x140012049c0?, 0x14001379bc0, 0x14001b48b50?, 0x140018da2d0, 0x14001205260)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1009 +0x31c
github.com/open-policy-agent/opa/v1/topdown.evalTerm.eval({0x140002c3800, 0x140018da2d0, 0x14000377d58, 0x140018da2d0, 0x14001379bc0, 0x140018da2d0, {0x14001035320, 0x4, 0x4}, 0x4}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3608 +0x17c
github.com/open-policy-agent/opa/v1/topdown.evalTerm.next({0x140002c3800, 0x140018da2d0, 0x14000c6fd88, 0x140018da2d0, 0x14001379bc0, 0x140018da2d0, {0x14001035320, 0x4, 0x4}, 0x3}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3635 +0xd4
github.com/open-policy-agent/opa/v1/topdown.evalTerm.eval({0x140002c3800, 0x140018da2d0, 0x14000c6fd88, 0x140018da2d0, 0x14001379bc0, 0x140018da2d0, {0x14001035320, 0x4, 0x4}, 0x3}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3618 +0xdc
github.com/open-policy-agent/opa/v1/topdown.evalTerm.next({0x140002c3800, 0x140018da2d0, 0x14000c6fe00, 0x140018da2d0, 0x14001379bc0, 0x140018da2d0, {0x14001035320, 0x4, 0x4}, 0x2}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3635 +0xd4
github.com/open-policy-agent/opa/v1/topdown.evalTerm.eval({0x140002c3800, 0x140018da2d0, 0x14000c6fe00, 0x140018da2d0, 0x14001379bc0, 0x140018da2d0, {0x14001035320, 0x4, 0x4}, 0x2}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3618 +0xdc
github.com/open-policy-agent/opa/v1/topdown.evalTerm.next({0x140002c3800, 0x140018da2d0, 0x14000c6fe60, 0x140018da2d0, 0x14001379bc0, 0x140018da2d0, {0x14001035320, 0x4, 0x4}, 0x1}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3635 +0xd4
github.com/open-policy-agent/opa/v1/topdown.evalTerm.eval({0x140002c3800, 0x140018da2d0, 0x14000c6fe60, 0x140018da2d0, 0x14001379bc0, 0x140018da2d0, {0x14001035320, 0x4, 0x4}, 0x1}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3618 +0xdc
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyRef(0x140002c3800, 0x10469750c?, 0x14001379bc0, 0x140018da2d0, 0x140018da2d0, 0x14001205260)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1225 +0x214
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyValues(0x140002c3800, 0x14001379bc0, 0x14000fe0330, 0x140018da2d0, 0x140018da2d0, 0x14001205260)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1121 +0x178
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunify(0x140002c3800, 0x10511def6?, 0x14000fe0330, 0x14000f416c0?, 0x140018da2d0, 0x14001205260)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:990 +0x1cc
github.com/open-policy-agent/opa/v1/topdown.(*eval).unify(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:979
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalStep(0x140002c3800, 0x14000f416c0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:498 +0x28c
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalExpr(0x140002c3800, 0x14001a27440)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:407 +0x21c
github.com/open-policy-agent/opa/v1/topdown.(*eval).next(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:208
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalExpr.func2(0x140002c3800)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:408 +0x30
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalStep.func6()
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:503 +0x28
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyValues(0x140002c3800, 0x14000fe0210, 0x14001205248, 0x140018da2d0, 0x140018da2d0, 0x14001205218)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1152 +0x364
...179 frames elided...
github.com/open-policy-agent/opa/v1/topdown.evalTerm.eval({0x140002c3600, 0x140017c2ff0, 0x14000c6fe00, 0x140017c2ff0, 0x140011a2648, 0x140017c2ff0, {0x140010b1d20, 0x3, 0x4}, 0x2}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3618 +0xdc
github.com/open-policy-agent/opa/v1/topdown.evalTerm.next({0x140002c3600, 0x140017c2ff0, 0x14000c6fe60, 0x140017c2ff0, 0x140011a2648, 0x140017c2ff0, {0x140010b1d20, 0x3, 0x4}, 0x1}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3635 +0xd4
github.com/open-policy-agent/opa/v1/topdown.evalTerm.eval({0x140002c3600, 0x140017c2ff0, 0x14000c6fe60, 0x140017c2ff0, 0x140011a2648, 0x140017c2ff0, {0x140010b1d20, 0x3, 0x4}, 0x1}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3618 +0xdc
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyRef(0x140002c3600, 0x105fae840?, 0x140011a2648, 0x140017c2ff0, 0x140017c2ff0, 0x140010caa80)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1225 +0x214
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyValues(0x140002c3600, 0x140011a2648, 0x14000cc0cd8, 0x140017c2ff0, 0x140017c2ff0, 0x140010caa80)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1121 +0x178
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunify(0x140002c3600, 0x10511def6?, 0x14000cc0cd8, 0x14000f1bda0?, 0x140017c2ff0, 0x140010caa80)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:990 +0x1cc
github.com/open-policy-agent/opa/v1/topdown.(*eval).unify(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:979
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalStep(0x140002c3600, 0x14000f1bda0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:498 +0x28c
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalExpr(0x140002c3600, 0x140019c0ea0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:407 +0x21c
github.com/open-policy-agent/opa/v1/topdown.(*eval).eval(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:356
github.com/open-policy-agent/opa/v1/topdown.evalVirtualComplete.evalValueRule({0x1400099b600, 0x14000f487c0, 0x140016ba030, 0x140010ca768, 0x140016ba030, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3440 +0x33c
github.com/open-policy-agent/opa/v1/topdown.evalVirtualComplete.evalValue.func1()
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3383 +0x104
github.com/open-policy-agent/opa/v1/topdown.withSuppressEarlyExit(0x1400192bcb8?)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:4166 +0x24
github.com/open-policy-agent/opa/v1/topdown.evalVirtualComplete.evalValue({0x1400099b600, 0x14000f487c0, 0x140016ba030, 0x140010ca768, 0x140016ba030, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3376 +0x154
github.com/open-policy-agent/opa/v1/topdown.evalVirtualComplete.eval({0x1400099b600, 0x14000f487c0, 0x140016ba030, 0x140010ca768, 0x140016ba030, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:3340 +0x254
github.com/open-policy-agent/opa/v1/topdown.evalVirtual.eval({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, 0x5}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2567 +0x1e0
github.com/open-policy-agent/opa/v1/topdown.evalTree.next({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, 0x14000300dc0, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2346 +0x258
github.com/open-policy-agent/opa/v1/topdown.evalTree.eval({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, 0x14000300dc0, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2301 +0xcc
github.com/open-policy-agent/opa/v1/topdown.evalTree.next({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, 0x14000300e10, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2352 +0x2a8
github.com/open-policy-agent/opa/v1/topdown.evalTree.eval({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, 0x14000300e10, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2301 +0xcc
github.com/open-policy-agent/opa/v1/topdown.evalTree.next({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, 0x14000300cd0, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2352 +0x2a8
github.com/open-policy-agent/opa/v1/topdown.evalTree.eval({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, 0x14000300cd0, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2301 +0xcc
github.com/open-policy-agent/opa/v1/topdown.evalTree.next({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, 0x14000300d20, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2352 +0x2a8
github.com/open-policy-agent/opa/v1/topdown.evalTree.eval({0x1400099b600, 0x140016ba030, 0x140010ca768, 0x140016ba030, 0x14000300d20, {0x14000dae240, 0x5, 0x8}, {0x140017c2f90, 0x5, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2301 +0xcc
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyRef(0x1400099b600, 0x105fae840?, 0x140010ca768, 0x140016ba030, 0x140016ba030, 0x140006b0600)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1194 +0x320
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyValues(0x1400099b600, 0x140010ca7f8, 0x140010ca768, 0x140016ba030, 0x140016ba030, 0x140006b0600)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1110 +0xd0
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunify(0x1400099b600, 0x5?, 0x140010ca768, 0x140016ba030?, 0x140016ba030, 0x140006b0600)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:990 +0x1cc
github.com/open-policy-agent/opa/v1/topdown.(*eval).unify(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:979
github.com/open-policy-agent/opa/v1/topdown.evalTree.leaves({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300dc0, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2487 +0x4b4
github.com/open-policy-agent/opa/v1/topdown.evalTree.extent({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300dc0, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2439 +0xc4
github.com/open-policy-agent/opa/v1/topdown.evalTree.finish({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300dc0, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2316 +0xc0
github.com/open-policy-agent/opa/v1/topdown.evalTree.eval({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300dc0, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2295 +0x144
github.com/open-policy-agent/opa/v1/topdown.evalTree.next({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300e10, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2352 +0x2a8
github.com/open-policy-agent/opa/v1/topdown.evalTree.eval({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300e10, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2301 +0xcc
github.com/open-policy-agent/opa/v1/topdown.evalTree.next({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300cd0, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2352 +0x2a8
github.com/open-policy-agent/opa/v1/topdown.evalTree.eval({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300cd0, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2301 +0xcc
github.com/open-policy-agent/opa/v1/topdown.evalTree.next({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300d20, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2352 +0x2a8
github.com/open-policy-agent/opa/v1/topdown.evalTree.eval({0x1400099b600, 0x140016ba030, 0x14000c6ea68, 0x140016ba030, 0x14000300d20, {0x14000904c40, 0x4, 0x4}, {0x140003a2700, 0x4, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:2301 +0xcc
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyRef(0x1400099b600, 0x105fae840?, 0x14000c6ea68, 0x140016ba030, 0x140016ba030, 0x14000c70000)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1194 +0x320
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunifyValues(0x1400099b600, 0x14000c6ea68, 0x14000c6eaf8, 0x140016ba030, 0x140016ba030, 0x14000c70000)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:1121 +0x178
github.com/open-policy-agent/opa/v1/topdown.(*eval).biunify(0x1400099b600, 0x10511def6?, 0x14000c6eaf8, 0x14000ed4da0?, 0x140016ba030, 0x14000c70000)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:990 +0x1cc
github.com/open-policy-agent/opa/v1/topdown.(*eval).unify(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:979
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalStep(0x1400099b600, 0x14000ed4da0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:498 +0x28c
github.com/open-policy-agent/opa/v1/topdown.(*eval).evalExpr(0x1400099b600, 0x14000ed4d90)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:407 +0x21c
github.com/open-policy-agent/opa/v1/topdown.(*eval).eval(...)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:356
github.com/open-policy-agent/opa/v1/topdown.(*eval).Run(0x140016ba0f0?, 0x10512c9f4?)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/eval.go:144 +0xc0
github.com/open-policy-agent/opa/v1/topdown.(*Query).Iter(0x14001b536c0, {0x106001040, 0x140016f8190}, 0x140003a26e0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/topdown/query.go:589 +0xb50
github.com/open-policy-agent/opa/v1/rego.(*Rego).eval(0x14000b36a88, {0x106001040, 0x140016f8190}, 0x1400025dba0)
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/rego/rego.go:2222 +0x6d0
github.com/open-policy-agent/opa/v1/rego.PreparedEvalQuery.Eval({{0x14000b36a88?, 0x140009f4be8?}}, {0x106001040, 0x140016f8190}, {0x1400005bdb8?, 0x0?, 0x0?})
        /home/runner/go/pkg/mod/github.com/open-policy-agent/opa@v1.1.0/v1/rego/rego.go:505 +0xf8
github.com/styrainc/regal/pkg/linter.Linter.lintWithRegoRules.func1({0x16b7d6ec3, 0x26})
        /home/runner/work/regal/regal/pkg/linter/linter.go:816 +0x4b0
created by github.com/styrainc/regal/pkg/linter.Linter.lintWithRegoRules in goroutine 1
        /home/runner/work/regal/regal/pkg/linter/linter.go:788 +0x2bc
```

--- a/pkg/builtins/builtins.go
+++ b/pkg/builtins/builtins.go
@@ -3,6 +3,7 @@ package builtins
 
 import (
 	"bytes"
+	"fmt"
 	"strings"
 
 	"github.com/open-policy-agent/opa/v1/ast"
@@ -150,12 +151,31 @@ func RegalIsFormatted(_ rego.BuiltinContext, input *ast.Term, options *ast.Term)
 	popts := ast.ParserOptions{ProcessAnnotation: false, RegoVersion: regoVersion}
 	source := util.StringToByteSlice(string(inputStr))
 
-	result, err := format.SourceWithOpts("", source, format.Opts{RegoVersion: regoVersion, ParserOptions: &popts})
+	result, err := formatRego(source, format.Opts{RegoVersion: regoVersion, ParserOptions: &popts})
 	if err != nil {
 		return nil, err
 	}
 
 	return ast.InternedBooleanTerm(bytes.Equal(source, result)), nil
+}
+
+func formatRego(source []byte, opts format.Opts) (result []byte, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			switch r := r.(type) {
+			case string:
+				err = fmt.Errorf("error formatting: %s", r)
+			case error:
+				err = r
+			default:
+				err = fmt.Errorf("error formatting: %v", r)
+			}
+		}
+	}()
+
+	result, err = format.SourceWithOpts("", source, opts)
+
+	return
 }
 
 // TestContextBuiltins returns the list of builtins as expected by the test runner.

